File: codegen/results/v001/apple-foundationmodel/20260409_105028/typescript/005_unit_test_writer/solution.ts
// Define the calculateDiscount function
function calculateDiscount(price: number, quantity: number): number {
  if (quantity < 10) return price * quantity;
  else if (quantity < 50) return price * quantity * 0.9;
  else return price * quantity * 0.75;
}

// Test cases including boundary conditions
const testCases = [
  {
    description: "Test with quantity 9 (below 10)",
    inputs: { price: 100, quantity: 9 },
    expectedOutput: 900,
  },
  {
    description: "Test with quantity 10 (exactly 10)",
    inputs: { price: 100, quantity: 10 },
    expectedOutput: 1000,
  },
  {
    description: "Test with quantity 49 (just below 50)",
    inputs: { price: 100, quantity: 49 },
    expectedOutput: 4900,
  },
  {
    description: "Test with quantity 50 (exactly 50)",
    inputs: { price: 100, quantity: 50 },
    expectedOutput: 5000,
  },
];

// Function to run tests and print results
function runTests(testCases: Array<{ description: string; inputs: { price: number; quantity: number }; expectedOutput: number }>): void {
  for (const { description, inputs, expectedOutput } of testCases) {
    const result = calculateDiscount(inputs.price, inputs.quantity);
    if (result === expectedOutput) {
      console.log(`PASS: ${description}`);
    } else {
      console.log(`FAIL: ${description} (expected: ${expectedOutput}, got: ${result})`);
    }
  }
}

// Run the tests
runTests(testCases);

File: codegen/results/v001/apple-foundationmodel/20260409_105028/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func testCalculateDiscount() {
	// Test case 1: Quantity 9
	price := 100.0
	quantity := 9
	expected := price * float64(quantity)
	result := calculateDiscount(price, quantity)
	if result == expected {
		fmt.Printf("PASS: Quantity 9\n")
	} else {
		fmt.Printf("FAIL: Quantity 9\nExpected: %f, got: %f\n", expected, result)
	}

	// Test case 2: Quantity 10
	price := 100.0
	quantity := 10
	expected := price * float64(quantity)
	result := calculateDiscount(price, quantity)
	if result == expected {
		fmt.Printf("PASS: Quantity 10\n")
	} else {
		fmt.Printf("FAIL: Quantity 10\nExpected: %f, got: %f\n", expected, result)
	}

	// Test case 3: Quantity 49
	price := 100.0
	quantity := 49
	expected := price * float64(quantity) * 0.9
	result := calculateDiscount(price, quantity)
	if result == expected {
		fmt.Printf("PASS: Quantity 49\n")
	} else {
		fmt.Printf("FAIL: Quantity 49\nExpected: %f, got: %f\n", expected, result)
	}

	// Test case 4: Quantity 50
	price := 100.0
	quantity := 50
	expected := price * float64(quantity) * 0.9
	result := calculateDiscount(price, quantity)
	if result == expected {
		fmt.Printf("PASS: Quantity 50\n")
	} else {
		fmt.Printf("FAIL: Quantity 50\nExpected: %f, got: %f\n", expected, result)
	}
}

func main() {
	testCalculateDiscount()
}